source/_integrations/yeelight.markdown

```diff
@@ -253,6 +253,15 @@ Turns the light on to the specified brightness and sets a timer to turn it back
 
 | Service data attribute    | Optional | Description                                                                                 |
 |---------------------------|----------|---------------------------------------------------------------------------------------------|
-| `entity_id`               |       no | Only act on specific lights.                                                              |
+| `entity_id`               |       no | Only act on specific lights.                                                                |
 | `minutes`                 |       no | The minutes to wait before automatically turning the light off.                             |
 | `brightness`              |       no | The brightness value to set (1-100).                                                        |
+
+### Service `yeelight.set_music_mode`
+
+Enable or disable music_mode.
+
+| Service data attribute    | Optional | Description                                                                                 |
+|---------------------------|----------|---------------------------------------------------------------------------------------------|
+| `entity_id`               |       no | Only act on specific lights.                                                                |
```

Comment:
This service uses the `async_register_entity_service`, which means that `entity_id` is optional.

```suggestion
| `entity_id`               |      yes | Only act on specific lights.                                                                |
```

Reply:
I've tried this without entity_id and got following error:

```
2020-12-29 20:56:54 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [140135236333376] sequence item 0: expected str instance, Optional found
Traceback (most recent call last):
  File "/home/omen/PycharmProjects/home-assistant/homeassistant/components/websocket_api/commands.py", line 135, in handle_call_service
    await hass.services.async_call(
  File "/home/omen/PycharmProjects/home-assistant/homeassistant/core.py", line 1399, in async_call
    processed_data = handler.schema(service_data)
  File "/home/omen/PycharmProjects/home-assistant/venv/lib/python3.8/site-packages/voluptuous/validators.py", line 218, in __call__
    return self._exec((Schema(val) for val in self.validators), v)
  File "/home/omen/PycharmProjects/home-assistant/venv/lib/python3.8/site-packages/voluptuous/validators.py", line 337, in _exec
    v = func(v)
  File "/home/omen/PycharmProjects/home-assistant/venv/lib/python3.8/site-packages/voluptuous/schema_builder.py", line 272, in __call__
    return self._compiled([], data)
  File "/home/omen/PycharmProjects/home-assistant/venv/lib/python3.8/site-packages/voluptuous/schema_builder.py", line 817, in validate_callable
    return schema(data)
  File "/home/omen/PycharmProjects/home-assistant/homeassistant/helpers/config_validation.py", line 140, in validate
    raise vol.Invalid("must contain at least one of {}.".format(", ".join(keys)))
TypeError: sequence item 0: expected str instance, Optional found
```

Reply:
No, you can target an area or device ID alternatively.

Reply:
I see thanks ! Changed as requested. In this case, should I also change other service calls descriptions ? I think all now use async_register_entity_service but in docs all are described as not optional, regarding entity_id. If yes id can be changed in this PR, or in new one ?

Reply:
If all use the same service, feel free to adjust the docs in this PR.

Reply:
Adjusted docs.